colibri/src/icons.rs

```diff
@@ -119,6 +138,86 @@ async fn query_token_icon_and_extension(
     None
 }
 
+async fn query_uniswap_v3_position_icon(
+    chain_id: u64,
+    token_id: &str,
+    contract_address: &str,
+    globaldb: &globaldb::GlobalDB,
+) -> Option<(Bytes, &'static str)> {
+    let blockchain = SupportedBlockchain::from_chain_id(chain_id)?;
+    let rpc_endpoints = globaldb.get_rpc_nodes(blockchain).await.ok()?;
+    let contract_address = match Address::parse_checksummed(contract_address, None) {
+        Ok(address) => address,
+        Err(_e) => {
+            return None;
+        }
+    };
+    let token_id: U256 = token_id.parse().unwrap();
+
+    // Try all RPC endpoints until one succeeds
+    for rpc_url in rpc_endpoints {
+        let provider = match rpc_url.parse().map(|url| ProviderBuilder::new().on_http(url)) {
+            Ok(provider) => provider,
+            Err(e) => {
+                error!("Failed to setup provider for {}: {}", rpc_url, e);
+                continue;
+            }
+        };
+
+        let contract = IERC721Metadata::new(contract_address, provider);
+        let token_uri = match contract.tokenURI(token_id).call().await {
+            Ok(result) => result._0,
+            Err(e) => {
+                error!("RPC call failed on {}: {}", rpc_url, e);
+                continue;
+            }
+        };
+        let base64_str = match token_uri.strip_prefix("data:application/json;base64,") {
+            Some(data) => data,
+            None => {
+                error!("Invalid token URI format from {}", rpc_url);
+                continue;
```

Comment:
do you want to continue here? This means that you got a response but the contract is badly formated. Any other node would return valid information?

Reply:
errr... not really, it can be other reasons like the rpc failing and not the contract.

Reply:
Isnt' tthere a way to distinguish between the error types? There should be.

If it's a badly formatted contract then stop. If the rpc node is failing then continue and try another one